tests/unit/mail_test.sh

```diff
@@ -204,6 +214,32 @@ function test_validate_email_list()
   assert_equals_helper 'Expected a success' "$LINENO" "$ret" 0
 }
 
+function test_count_patches()
+{
+  cd "$FAKE_GIT" || {
+    ret="$?"
+    fail "($LINENO): Failed to move to fake git dir"
+    exit "$ret"
+  }
+
+  local output
+  local expected
+
+  output="$(count_patches "$FIRST_TEST_PATCHES_DIR")"
+  expected=2
+  assert_equals_helper 'Expected two patches' "$LINENO" "$output" "$expected"
+
+  output="$(count_patches "$SECOND_TEST_PATCHES_DIR")"
+  expected=1
+  assert_equals_helper 'Expected two patches' "$LINENO" "$output" "$expected"
```

Comment:
You should add other tests that validate:
1. There is not patch.
2. The target file is not a patch.